### PR TITLE
Added Matlab implementation of SNS-IK solvers

### PR DESCRIPTION
**Pull request by MatthewPeterKelly**
This commit adds a few functions that implement various versions
of the SNS-IK solver algorithms, along with a set of unit tests.
This code is primarily used for reference and debugging, rather
than as a well-packaged product.

### CODE REVIEW

**Comment by MatthewPeterKelly** [comment]
Following up on the naming question:  I propose that we keep the names as they are for now, but then come up with a better (and standardized) naming convention once we get a larger set of methods working.

File: sns_ik_lib/matlab/snsIk_acc_rr.m [hunk 1]
```diff
@@ -0,0 +1,122 @@
+function [ddq, s, exitCode] = snsIk_acc_rr(ddqLow, ddqUpp, ddxGoal, J, dJdq)
+% [ddq, s, exitCode] = snsIk_acc_rr(ddqLow, ddqUpp, ddxGoal, J, dJdq)
+%
+% This function implements the basic QP version of the SNS-IK velocity
```
Comment (robodreamer) [inline sns_ik_lib/matlab/snsIk_acc_rr.m]: This function description needs to be updated.

File: sns_ik_lib/matlab/snsIk_acc_rr.m [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+%   ddqLow: lower limit for joint acceleration
+%   ddqUpp: upper limit for joint acceleration
+%   ddxGoal: task-space velocity
+%   J: jacoabian mapping joint space to task space
+%   dJdq = dJ * dq
+%       dJ = time-derivative of the task jacobian
+%       dq = current joint velocity
+%
+% OUTPUTS:
+%   ddq = joint velocity solution with maximum task scale factor
```
Comment (robodreamer) [inline sns_ik_lib/matlab/snsIk_acc_rr.m]: joint acceleration solution

File: sns_ik_lib/matlab/snsIk_acc_rr.m (hunk 1 above)
Comment (robodreamer) [reply sns_ik_lib/matlab/snsIk_acc_rr.m]: velocity -> acceleration

Comment (robodreamer) [reply sns_ik_lib/matlab/snsIk_acc_rr.m]: QP needs to be removed

File: sns_ik_lib/matlab/snsIk_acc_QP.m [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+%   dJdq = dJ * dq
+%       dJ = time-derivative of the task jacobian
+%       dq = current joint velocity
+%   alpha: trade-off between minimum-joint velocity and maximum task scale
+%      default: alpha = 1e-3
+%      small alpha: favor solutions that maximize task scale
+%      large alpha: favor minimum-joint acceleration solutions
+%
+% OUTPUTS:
+%   ddq = joint velocity solution with maximum task scale factor
```
Comment (robodreamer) [inline sns_ik_lib/matlab/snsIk_acc_QP.m]: velocity solution -> acceleration solution

File: sns_ik_lib/matlab/TEST_acc_ik_solvers.m [hunk 4]
```diff
@@ -0,0 +1,18 @@
+% This script sets up an runs a set of unit tests on the acceleration IK solvers.
```
Comment (IanTheEngineer) [inline sns_ik_lib/matlab/TEST_acc_ik_solvers.m]: We'll need a license file in this folder, and an abbreviated license with a Rethink copyright at the top of each matlab file.

File: sns_ik_lib/matlab/snsIk_acc_rr.m (hunk 1 above)
Comment (MatthewPeterKelly) [reply sns_ik_lib/matlab/snsIk_acc_rr.m]: The hazards of cut and paste!  I fixed it. I think.

File: sns_ik_lib/matlab/LICENSE.txt [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "{}"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright {2016} {Fabrizio Flacco}
```
Comment (IanTheEngineer) [inline sns_ik_lib/matlab/LICENSE.txt]: Please add in:
``` 
Copyright 2016-2018 Rethink Robotics Inc.
```
After Fabrizio's copyright. We should probably do the same thing at the top level License

File: sns_ik_lib/matlab/TEST_acc_ik_solvers.m [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+%
+%
+% LICENSE:
```
Comment (IanTheEngineer) [inline sns_ik_lib/matlab/TEST_acc_ik_solvers.m]: License is redundant, you can remove that line, and un-indent the rest of the license

File: sns_ik_lib/matlab/TEST_vel_ik_solvers.m [hunk 7]
```diff
@@ -0,0 +1,39 @@
+% Run a simple unit test on the matlab implementation for the various
```
Comment (IanTheEngineer) [inline sns_ik_lib/matlab/TEST_vel_ik_solvers.m]: Unless it affects how matlab builds documentation (?), i would move the description below the Copyright and License

File: sns_ik_lib/matlab/runTest_snsIk_acc.m [hunk 8]
```diff
@@ -0,0 +1,123 @@
+function result = runTest_snsIk_acc(solver, nTest, optTol, cstTol, fid)
```
Comment (IanTheEngineer) [inline sns_ik_lib/matlab/runTest_snsIk_acc.m]: Abbrev copyright above

File: sns_ik_lib/matlab/runTest_snsIk_vel.m [hunk 9]
```diff
@@ -0,0 +1,120 @@
+function result = runTest_snsIk_vel(solver, nTest, optTol, cstTol, fid)
```
Comment (IanTheEngineer) [inline sns_ik_lib/matlab/runTest_snsIk_vel.m]: Abbrev copyright

File: sns_ik_lib/matlab/LICENSE.txt (hunk 5 above)
Comment (MatthewPeterKelly) [reply sns_ik_lib/matlab/LICENSE.txt]: done.

File: sns_ik_lib/matlab/TEST_acc_ik_solvers.m (hunk 6 above)
Comment (MatthewPeterKelly) [reply sns_ik_lib/matlab/TEST_acc_ik_solvers.m]: fixed

File: sns_ik_lib/matlab/TEST_vel_ik_solvers.m (hunk 7 above)
Comment (MatthewPeterKelly) [reply sns_ik_lib/matlab/TEST_vel_ik_solvers.m]: Matlab uses the first comment block after a function definition as the documentation that is auto-generated by the help command. I looked at how the license is included in the Matlab source code, and modified our license to be formatted similarly: leave one blank line after the help file and then include the license as a comment.

File: sns_ik_lib/matlab/runTest_snsIk_acc.m [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+%       OUT: exitCode  (1 == success)
+%   nTest: how many tests to run?
+%   optTol: optimality tolerance
+%   cstTol: feasibility tolerance
+%   fid: output stream for test logs (1 == command prompt)
+%
+% OUTPUTS:
+%   result: struct with test results
+
+% Copyright 2018 Rethink Robotics
```
Comment (IanTheEngineer) [inline sns_ik_lib/matlab/runTest_snsIk_acc.m]: Sorry for the nitpick, but this should be at the top

Comment (IanTheEngineer) [reply sns_ik_lib/matlab/runTest_snsIk_acc.m]: for each of the function files, unless it breaks matlab

**Pull request merged**